osp/core/utils/general.py

```diff
@@ -214,165 +295,255 @@ def get_custom_datatype_triples():
-    from osp.core.session.buffers import BufferContext
-    from osp.core.ontology.cuba import rdflib_cuba
-    if isinstance(json_doc, str):
-        json_doc = json.loads(json_doc)
-    session = session or Cuds._session
-    buffer_context = buffer_context or BufferContext.USER
-    g = json_to_rdf(json_doc, Graph())
-    # only return first
-    first = g.value(rdflib_cuba._serialization, RDF.first)
-    first_uid = None
-    if first:  # return the element marked as first later
-        first_uid = uuid.UUID(hex=first)
-        g.remove((rdflib_cuba._serialization, RDF.first, None))
-    deserialized = import_rdf(
-        graph=g,
-        session=session,
-        buffer_context=buffer_context,
-        return_uid=first_uid
-    )
-    return deserialized
-
-
-def import_rdf_file(path, format="xml", session=None, buffer_context=None):
-    """Import rdf from file.
+    if not any(bool(x) for x in (path, file)):
+        raise ValueError('Specify a path or a file (a file-like object, '
+                         'a string, or a json document dictionary) '
+                         'to read from.')
+    elif sum(bool(x) for x in (path, file)) > 1:
+        raise ValueError('Specify either a path or a file (a file-like object,'
+                         ' a string, or a json document dictionary) '
+                         'to read from.')
+    elif format not in ('json', 'xml', 'ttl'):
+        raise ValueError(f'Unsupported format {format}. The supported formats '
+                         f'are: {", ".join(("json", "xml", "ttl"))}. '
+                         f'Please check the docstring of this function for '
+                         f'more information.')
+
+    if path:
+        if not pathlib.Path(path).is_file():
+            raise Exception(f'{path} is not a file.')
+        with open(path, 'r') as file:
+            contents = file.read()
+            file_like = False
+    elif file:
+        if type(file) in (str, dict):
+            if type(file) is dict and format != 'json':
+                raise TypeError('Not a json document.')
+            contents = file
+            file_like = False
+        else:
+            if '__read__' not in file.__dir__():
+                raise TypeError(f'{file} is not a file-like object.')
+            contents = file
+            file_like = True
+
+    from osp.core.session.core_session import core_session
+    session = session or core_session
+    if format == 'json':
+        results = deserialize_cuds_object(contents, session=session,
+                                          buffer_context=None)
+    elif format in ('xml', 'ttl'):
+        results = import_rdf_file(contents if file_like
+                                  else io.StringIO(contents),
+                                  format=format, session=session,
+                                  buffer_context=None)
+    return results
+
+
+def export(item: Optional = None,
```

Comment:
Instead of taking the core session by default, use the class session (under the `with` statement)

Reply:
Done